### PR TITLE
Add RapidJSON to third_party folder

### PR DESCRIPTION
**Pull request by karthikreddy09**
Signed-off-by: The Mathworks Inc <Roy.Lurie@mathworks.com>

### CODE REVIEW

File: third_party/README [hunk 1]
```diff
@@ -138,3 +138,10 @@ Patches
 
  * Suppress CMake warning about policy CMP0048
  * Suppress tr1 namespace deprecation warnings
+
+ RapidJSON
+ ---------
+
+ MIT License
+
+ https://github.com/Tencent/rapidjson/tree/version1.1.0
```
Comment (saschazelzer) [inline third_party/README]: I would prefer the same style as above: Pointing to the release page and state the version number in a separate line below.

Comment (karthikreddy09) [reply third_party/README]: Updated to match existing style

Comment (karthikreddy09) [reply third_party/README]: @saschazelzer Could you please take a look at the latest changes and provide your feedback. I have other changes I want to publish, which depend on this PR.

**Comment by karthikreddy09** [comment]
fixes #335

**Pull request merged**